lib/nepali_calendar/bs_calendar.rb

```diff
@@ -54,6 +54,75 @@ def end_of_month
       NepaliCalendar::BsCalendar.travel days, date
     end
 
+      # Returns the bs date that is the beginning of the provided fiscal_year.
+      def self.beginning_of_fiscal_year_in_bs(fiscal_year)
+        start_year = fiscal_year.to_s.slice(0, 2).prepend('20')
+        start_date = NepaliCalendar::Calendar.new(nil, {year: start_year, month: 4, day: 1}) #start date of fiscal year
+      end
+
+
+      # Returns the bs date that is the end of the provided fiscal_year.
+      def self.end_of_fiscal_year_in_bs(fiscal_year)
+        end_year = fiscal_year.to_s.slice(2, 2).prepend('20')
+        end_date = NepaliCalendar::Calendar.new(nil, {year: end_year, month: 3, day: NepaliCalendar::BS[end_year.to_i][3]}) #end date of fiscal year
+      end
+
+
+      # Returns the ad date that is the beginning of the provided fiscal_year.
+      def self.beginning_of_fiscal_year_in_ad(fiscal_year)
+        bs_start_date= beginning_of_fiscal_year_in_bs(fiscal_year)
+        NepaliCalendar::AdCalendar.bs_to_ad(bs_start_date.year, bs_start_date.month, bs_start_date.day) #stard date in AD of the corresponding nepali date
+      end
+
+      # Returns the ad date that is the end of the provided fiscal_year.
+      def self.end_of_fiscal_year_in_ad(fiscal_year)
+        ad_end_date = end_of_fiscal_year_in_bs(fiscal_year)
+        NepaliCalendar::AdCalendar.bs_to_ad(ad_end_date.year, ad_end_date.month, ad_end_date.day)  #end date in AD of the fiscal year
+      end
```

Comment:
Let's get rid of this method. Now that I think about it, it adds unnecessary complexity. 

To achieve same, we could rather use
1. Get fy beginning in bs date
2. Convert the bs date to ad date.